src/api/registerApplicationResourceResolver.ts

```diff
@@ -17,6 +20,10 @@ export function registerApplicationResourceResolver(id: string, resolver: AppRes
     applicationResourceResolvers[id] = resolver;
     ext.emitters.onDidRegisterResolver.fire(resolver);
 
+    const compat = new CompatibleBranchDataProvider(resolver, 'azureResourceGroups.loadMore' /** TODO: what is the correct value for this? */);
+
+    ext.v2.api.registerApplicationResourceBranchDataProvider(id, compat as unknown as BranchDataProvider<ApplicationResource, CompatibleBranchDataItem>);
```

Comment:
This is now incompatible with the change to register BDPs by type. I'm a little reluctant to go ahead with that, as we'd need to do a built-in BDP for each of the unsupported resource types, whereas before we had the `matchesResource` method that could just check, and so there was a single BDP for all of them. @philliphoff what do you think?

Reply:
> This is now incompatible with the change to register BDPs by type

I'm not sure what you mean by this. I was planning on changing the IDs of the resolvers in the client extensions to match the `AzExtResourceType`.

Ex: https://github.com/microsoft/vscode-azurefunctions/blob/1d2514971b53806b9352844eadfc350cf31d4a56/src/extension.ts#L98

Reply:
Your change makes `registerApplicationResourceBranchDataProvider` accept an `AzExtResourceType` as the first parameter, not an ID, so line 25 here would need to change. But it carries the disadvantage that no BDP can represent multiple types.

Reply:
> we'd need to do a built-in BDP for each of the unsupported resource types

I think that's covered by the `defaultBranchDataProvider` argument in `BranchDataProviderManager`. It takes a BDP and will use that if no matching BDPs are found.

https://github.com/microsoft/vscode-azureresourcegroups/blob/alex/compat-shim-changes-1/src/tree/v2/providers/BranchDataProviderManager.ts#L14

Reply:
> But it carries the disadvantage that no BDP can represent multiple types.

I thought we'd just register the same instance of the BDP multiple times to cover that scenario.

Reply:
I see, I didn't realize that's how the default BDP worked. I think for more narrowly-scoped cases, where one BDP is handling e.g. 2 types, it's (probably) fine to reregister the same one several times (maybe there are some issues when it comes to change notification events etc.). What I wanted to avoid was ~30 registrations for unsupported types, but the way the default is implemented, that will not be necessary.

Reply:
Still, don't we need to change `id` in this line to the `AzExtResourceType`?

Reply:
I'm expecting the `AzExtResourceType` to be passed in by the client extensions, so `id` would be an `AzExtResourceType` already. I updated the param name and type from `id: string` to `type: AzExtResourceType`.